File: billing-domain/src/main/java/com/example/billing/domain/budget/BudgetAlertStatus.java
package com.example.billing.domain.budget;

/**
 * BudgetAlertRule 라이프사이클.
 *
 * <pre>
 *   ACTIVE ──pause()──▶ PAUSED
 *      ▲                    │
 *      └────resume()────────┘
 * </pre>
 *
 * <p>삭제 (DELETED) 는 별도 status 가 아니라 row 삭제로 처리. audit 가 필요해지면 도입.</p>
 */
public enum BudgetAlertStatus {
    ACTIVE,
    PAUSED
}

File: billing-domain/src/main/kotlin/com/example/billing/domain/budget/BudgetAlertStatus.kt
package com.example.billing.domain.budget

/**
 * BudgetAlertRule 라이프사이클.
 *
 * ```
 *   ACTIVE ──pause()──▶ PAUSED
 *      ▲                    │
 *      └────resume()────────┘
 * ```
 *
 * 삭제 (DELETED) 는 별도 status 가 아니라 row 삭제로 처리. audit 가 필요해지면 도입.
 */
enum class BudgetAlertStatus {
    ACTIVE,
    PAUSED,
}
